pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+"""Module containing pyvista implementation of vtkLight."""
+
+from enum import IntEnum
+
+import vtk
+from vtk import vtkLight
+
+import pyvista
+
+class LightType(IntEnum):
+    """An enumeration for the light types."""
+
+    HEADLIGHT = 1
+    CAMERA_LIGHT = 2
+    SCENE_LIGHT = 3
+
+    def __str__(self):
+        """Pretty name for a light type."""
+        return self.name.replace('_', ' ').title()
+
+class Light(vtkLight):
+    """Light class.
+
+    Parameters
+    ----------
+    position : list or tuple, optional
+        The position of the light. The interpretation of the position depends
+        on the type of the light.
+
+    color : list or tuple, optional
+        The color of the light. The ambient, diffuse and specular colors will
+        all be set to this color in creation.
+
+    light_type : string or int, optional
+        The type of the light. If a string, one of ``'headlight'``,
+        ``'camera light'`` or ``'scene light'``. If an int, one of 1, 2 or 3,
+        respectively. The class constants ``Light.HEADLIGHT``, ``Light.CAMERA_LIGHT``
+        and ``Light.SCENE_LIGHT`` are also available, respectively.
+
+        A headlight is attached to the camera, looking at its focal point along
+        the axis of the camera.
+        A camera light also moves with the camera, but it can occupy a general
+        position with respect to it.
+        A scene light is stationary with respect to the scene, as it does not
+        follow the camera. This is the default.
+    """
+
+    # TODO: better/more explanation for ``position``?
+
+    # pull in light type enum values as class constants
+    HEADLIGHT = LightType.HEADLIGHT
+    CAMERA_LIGHT = LightType.CAMERA_LIGHT
+    SCENE_LIGHT = LightType.SCENE_LIGHT
+    # TODO: is this design OK?
+
+    def __init__(self, position=None, color=None, light_type='scene light'):
+        """Initialize the light."""
+        super().__init__()
+
+        if position:
+            self.position = position
+
+        if color:
+            self.ambient_color = color
+            self.diffuse_color = color
+            self.specular_color = color
+            # TODO: make this a single call if that's implemented
+
+        if isinstance(light_type, str):
+            # be forgiving: ignore spaces and case
+            light_type_orig = light_type
+            type_normalized = light_type.replace(' ', '').lower()
+            mapping = {'headlight': LightType.HEADLIGHT,
+                       'cameralight': LightType.CAMERA_LIGHT,
+                       'scenelight': LightType.SCENE_LIGHT,
+                      }
+            light_type = mapping.get(type_normalized)
+            if light_type is None:
+                raise ValueError(f'Invalid ``light_type`` "{light_type_orig}"')
```

Comment:
How about something like(not tested):

```suggestion
            # be forgiving: ignore spaces and case
            type_normalized = light_type.replace(' ', '').lower()
            mapping = {str(e).replace(' ', '').lower(): e for e in LightType}
            try:
                light_type = mapping[type_normalized]
            except KeyError:
                raise ValueError(f'Invalid ``light_type`` "{light_type}"')
```

Reply:
I thought about automating it with something like that, but I was worried that it might make the intention harder to read and so the code harder to maintain.

As for LBYL vs EAFP: well, both can work ;) Not sure which one is more prevalent in the codebase, I'll try to see if there are other examples.